src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -542,33 +542,17 @@ private static bool CheckParameterRefEscape(SyntaxNode node, BoundParameter para
 
             // byval parameters can escape to method's top level.
             // others can be escape further, unless they are ref-like.
-            if (escapeTo == Binder.ExternalScope)
+            if (escapeTo == Binder.ExternalScope && parameterSymbol.RefKind == RefKind.None)
             {
-                if (parameterSymbol.RefKind == RefKind.None)
+                if (checkingReceiver)
                 {
-                    if (checkingReceiver)
-                    {
-                        Error(diagnostics, ErrorCode.ERR_RefReturnParameter2, parameter.Syntax, parameterSymbol.Name);
-                    }
-                    else
-                    {
-                        Error(diagnostics, ErrorCode.ERR_RefReturnParameter, node, parameterSymbol.Name);
-                    }
-                    return false;
+                    Error(diagnostics, ErrorCode.ERR_RefReturnParameter2, parameter.Syntax, parameterSymbol.Name);
                 }
-
-                if (parameterSymbol.Type?.IsByRefLikeType == true)
+                else
                 {
-                    if (checkingReceiver)
-                    {
-                        Error(diagnostics, ErrorCode.ERR_RefReturnParameter2, parameter.Syntax, parameterSymbol.Name);
```

Comment:
Do we still need those two error codes and corresponding messages?

Reply:
That codepath reused the same errors as for byval parameters, adding to the confusion. So, since we did not add a specific error for this., there is nothing to remove. There is probably a bug somewhere asking to improve quality of an error in this scenario. If such exists, it can be closed now.

---
In reply to: [141155238](https://github.com/dotnet/roslyn/pull/22303#discussion_r141155238) [](ancestors = 141155238)

Reply:
Closed [https://github.com/dotnet/roslyn/issues/22057](https://github.com/dotnet/roslyn/issues/22057)  - no longer relevant. The scenario is gone.

---
In reply to: [141166324](https://github.com/dotnet/roslyn/pull/22303#discussion_r141166324) [](ancestors = 141166324,141155238)